File: Tests/TestData/MultiFileCharacterization/VBToCSResults/ConvertVbUsingCSharpRefReturnOnly/VisualBasicUsesCSharpRefReturn/WithRefReturnStructure.cs

namespace VisualBasicUsesCSharpRefReturn
{
    public class WithRefReturnStructure
    {
        public void UseArr()
        {
            var arr = default(SomeStruct[]);
            var s = default(string);

            {
                ref var withBlock = ref arr[0];
                withBlock.P = s;
                s = withBlock.P;
            }
        }

        public void UseRefReturn()
        {
            var lst = default(CSharpRefReturn.RefReturnList<SomeStruct>);
            var s = default(string);

            {
                ref var withBlock = ref lst[0];
                withBlock.P = s;
                s = withBlock.P;
            }

            {
                ref var withBlock1 = ref lst.RefProperty;
                withBlock1.P = s;
                s = withBlock1.P;
            }
        }

        public struct SomeStruct
        {
            public string P { get; set; }
        }
    }
}

File: Tests/TestData/MultiFileCharacterization/SourceFiles/VisualBasicUsesCSharpRefReturn/WithRefReturnStructure.vb
Public Class WithRefReturnStructure
    Sub UseArr()
        Dim arr() As SomeStruct
        Dim s As String

        With arr(0)
            .P = s
            s = .P
        End With
    End Sub

    Sub UseRefReturn()
        Dim lst As CSharpRefReturn.RefReturnList(Of SomeStruct)
        Dim s As String

        With lst(0)
            .P = s
            s = .P
        End With

        With lst.RefProperty
            .P = s
            s = .P
        End With
    End Sub

    Structure SomeStruct
        Public Property P As String
    End Structure
End Class
